File: Channels/News_Channel/newsmainichi.rb
# Don't know how to get Mainichi working in Python, so we're using Ruby to parse it for now.
require 'nokogiri'
require 'open-uri'
# Or change to any other page
document = Nokogiri::HTML(open(ARGV[0]))
File.write("temp_mainichi", document)

File: Channels/News_Channel/newsmainichi.py
# Rewrote Mainichi in Python - TheMrIron2
import sys
import os
from bs4 import BeautifulSoup
os.chdir(r'C:\Python27') #insertyourpathere
html = open("page.html", 'r+') #changethehtmlfilenameifyouwant
soup = BeautifulSoup(html, 'html.parser')
with open("temp_mainichi.html", "w") as file:
    file.write(str(soup))
